modules/misunderstood/src/views/full/index.tsx

```diff
@@ -181,6 +228,14 @@ export default class MisunderstoodMainView extends React.Component<Props, State>
     }
   }
 
+  handleDateChange = async (dateRange: DateRange) => {
```

Comment:
I might be off here but I feel like this will trigger quite a lot of re-renders, like one for each setStateP (4). Can you optimize this for a single or double re-renders at max (1 loading state and 1 reload)

Reply:
Agreed, setEventIndex will also re-fetch the event that was just fetched... Can be a single setState 

`this.setState({ dateRange, events, selectedEventIndex: 0, selectedEvent: events?.[0] })`

Reply:
Yes it will update 4 times, i was trying to reuse functions already implemented in the component, i will change as requested